core/tests/puppeteer-acceptance-tests/user-utilities/exploration-editor-utils.ts

```diff
@@ -0,0 +1,288 @@
+   * @param {string} cardName - The name of the card to navigate to.
+   */
+  async navigateToCard(cardName: string): Promise<void> {
+    if (cardName === 'Introduction' && this.isViewportAtMobileWidth()) {
+      await this.clickOn(mobileStateGraphResizeButton);
+      await this.page.waitForSelector(mobileStateNodeSelector);
+      const elements = await this.page.$$(mobileStateNodeSelector);
+      await elements[3].click();
+    } else if (cardName === 'Introduction') {
+      await this.page.waitForSelector(stateNodeSelector);
+      const elements = await this.page.$$(stateNodeSelector);
+      await elements[0].click();
+    } else if (this.isViewportAtMobileWidth()) {
+      await this.clickOn(mobileStateGraphResizeButton);
+      await this.page.waitForSelector(mobileStateNodeSelector);
+      const elements = await this.page.$$(mobileStateNodeSelector);
+      if (cardName === 'Test Question ') {
+        await elements[3].click();
+      } else {
+        await elements[5].click();
+      }
+      await this.page.waitForNetworkIdle({idleTime: 700});
+    } else {
+      await this.clickOn(cardName);
+      await this.page.waitForNetworkIdle({idleTime: 700});
+    }
+  }
+
+  /**
+   * Function to add responses to the interactions.
+   * @param {string} response - response to be added.
+   */
+  async addResponseToTheInteraction(response: string): Promise<void> {
+    await this.type(floatFormInput, response);
+    await this.clickOn(feedbackEditorSelector);
+    await this.type(stateContentInputField, 'Correct Answer, You got that!');
+    await this.clickOn(correctAnswerInTheGroupSelector);
+    await this.clickOn(addNewResponseButton);
+    await this.page.waitForSelector(resonseModalHeaderSelector, {hidden: true});
+  }
+
+  /**
+   * Function to navigate to the preview tab.
+   */
+  async navigateToPreviewTab(): Promise<void> {
+    if (this.isViewportAtMobileWidth()) {
+      await this.clickOn(mobileNavbarDropdown);
+      await this.page.waitForSelector(mobileNavbarPane);
+      await this.clickOn(mobilePreviewTabButton);
+      await this.page.waitForNavigation();
+    } else {
+      await this.clickOn(previewTabButton);
+      await this.page.waitForNavigation();
+    }
+  }
+
+  /**
+   * Function to verify if the exploration is loading correctly in
+   * the preview tab or not via checking the content of the Introduction(first) card.
+   * @param {string} text - The expected introduction card text.
+   */
+  async expectCardContentToBe(text: string): Promise<void> {
+    await this.page.waitForSelector(stateConversationContent, {
+      visible: true,
+    });
+    const element = await this.page.$(stateConversationContent);
+    const introMessage = await this.page.evaluate(
+      element => element.textContent,
+      element
+    );
+    if (introMessage === text) {
+      showMessage(
+        'Preview is on the Introduction card and is loading correctly.'
```

Comment:
It's unclear from the name of this function that (a) it is being called in the context of the preview card, and (b) the text specifically refers to the Introduction card. Please make this utility function more general and name it clearly to indicate the scope in which it should be used.

Reply:
Addressed both the concerns and tried making the function more general.